examples/sim_fibo_ad_cp.cpp

```diff
@@ -206,6 +207,38 @@ try
                                   *props, deck, gravity[2], state);
     }
 
+    std::shared_ptr<BlackoilState> distributed_state=Dune::stackobject_to_shared_ptr(state);
```

Comment:
I am not quite comfortable with using the stackobject_to_shared_ptr() function. I do not doubt that it is correctly implemented, but it makes it easier to be confused about object lifetimes since it creates a shared_ptr with no ownership participation, essentially same as reference semantics. In my opinion the function should be limited to use in situations where we need to call a function taking a shared_ptr, the function could (semantically) have used a reference (not storing the pointer anywhere), yet there is no simple way to get rewrite the function in question. That situation is not present here, so I will propose a different solution:
```
BlackoilState distributed_state;
const bool must_distribute = grid->comm().size() > 1;
if (must_distribute) {
    // Initialise distributed_state etc.
    ...
}
...
BlackoilState& state_to_use = must_distribute ? distributed_state : state;
SimulatorReport fullReport = simulator.run(simtimer, state_to_use);
```